isis/tests/AlternativeTestCsmModel.cpp

```diff
@@ -0,0 +1,492 @@
+ */
+std::string AlternativeTestCsmModel::getParameterName(int index) const {
+  return AlternativeTestCsmModel::PARAM_NAMES[index];
+}
+
+
+/**
+ * Returns the sensor model parameter units at the provided index
+ * 
+ * @param index parameter unit index
+ * 
+ * @return std::string parameter units
+ */
+std::string AlternativeTestCsmModel::getParameterUnits(int index) const {
+  return AlternativeTestCsmModel::PARAM_UNITS[index];
+}
+
+
+/**
+ * True if the sensor model has sharable parameters. 
+ * 
+ * @return bool Always returns false
+ */
+bool AlternativeTestCsmModel::hasShareableParameters() const {
+  return false;
+}
+
+
+/**
+ * True if the sensor model parameter at the provided index is sharable.
+ * 
+ * @param index Parameter index
+ * 
+ * @return bool Always returns false
+ */
+bool AlternativeTestCsmModel::isParameterShareable(int index) const {
+  return false;
+}
+
+
+/**
+ * Returns the sharing criteria for the sensor model parameter at the provided index
+ * 
+ * @param index Parameter index
+ * 
+ * @return csm::SharingCriteria CSM sharing criteria for the parameter
+ */
+csm::SharingCriteria AlternativeTestCsmModel::getParameterSharingCriteria(int index) const {
+  return AlternativeTestCsmModel::PARAM_SHARING_CRITERIA[index];
+}
+
+
+/**
+ * Returns the sensor model parameter value at the provided index.
+ * 
+ * @param index Parameter index
+ * 
+ * @return double Value at provided index
+ */
+double AlternativeTestCsmModel::getParameterValue(int index) const {
+  return m_param_values[index];
+}
+
+
+/**
+ * Set the sensor model parameter at the provided index to the provided 
+ * value. 
+ * 
+ * @param index Parameter index
+ * @param value Value to set the parameter to
+ */
+void AlternativeTestCsmModel::setParameterValue(int index, double value) {
+  m_param_values[index] = value;
+}
+
+
+/**
+ * Returns the type of the sensor model parameter at the provided index.
+ * 
+ * @param index Parameter index
+ * 
+ * @return csm::param::Type Type of parameter
+ */
+csm::param::Type AlternativeTestCsmModel::getParameterType(int index) const {
+  return AlternativeTestCsmModel::PARAM_TYPES[index];
+}
+
+
+/**
+ * Does nothing. Minimal implementation for testing
+ * 
+ * @param index Parameter index
+ * @param pType Parameter type
+ */
+void AlternativeTestCsmModel::setParameterType(int index, csm::param::Type pType) {
+  // do nothing for test
+}
+
+
+/**
+ * Returns the covariance between the two sensor model parameters at the provided indicies. 
+ * Defaults to identity covariance matrix for testing. 
+ * 
+ * @param index1 First parameter index
+ * @param index2 Second parameter index
+ * 
+ * @return double Parameter covariance
+ */
+double AlternativeTestCsmModel::getParameterCovariance(int index1,
+                                            int index2) const {
+  // default to identity covariance matrix
+  if (index1 == index2) {
+    return 1.0;
+  }
+  return 0.0;
+                              }
```

Comment:
Misaligned curly bracket

Reply:
Wow that is misaligned! Thanks for catching this.